types/DatetimeLit.hpp

```diff
@@ -51,53 +54,70 @@ struct DateLit {
         + 1   // -
         + 2;  // Day
 
+  // Years should be between [-kMaxYear, +kMaxYear] inclusive both end.
+  static constexpr std::int32_t kMaxYear = 99999;
+  static constexpr std::uint8_t kBitsNeededForDay = 5u;
+  static constexpr std::uint8_t kBitsNeededForMonth = 4u;
+
   static DateLit Create(const std::int32_t _year,
                         const std::uint8_t _month,
                         const std::uint8_t _day) {
     DateLit date;
-    date.year = _year;
-    date.month = _month;
-    date.day = _day;
+    // Normalize year by adding kMaxYear value, because we try to
+    // encode signed year value into an unsigned integer.
```

Comment:
The check for leap year plays a role In the implementations for operator + and - for the date type. For instance, to perform an operation like ``-2000-29-01 + 13 months``, we need to know if the resultant year will be leap or not. It seems that this check for leap year will be performed on the normalized year instead of the original year.